src/backend/marsha/bbb/serializers.py

```diff
@@ -224,6 +225,28 @@ class Meta:  # noqa
         read_only=True, pk_field=serializers.CharField()
     )
 
+    @property
+    def max_upload_file_size(self):
+        """return the class room document max file size define in the settings.
+
+        The @property decorator is used to ease the use of @override_settings
+        in tests. Otherwise the setting is not changed and we can't easily test
+        an upload with a size higher than the one defined in the settings
+        """
+        return settings.CLASSROOM_DOCUMENT_SOURCE_MAX_SIZE
+
+    def to_internal_value(self, data):
+        """Validate if the size is coherent with django settings."""
+        if "size" in data and data.pop("size") > self.max_upload_file_size:
```

Comment:
So without `size` in the request there is no validation and I can create a document ?

Reply:
Actually, size is always sent by the frontend.
But I'll add a check, you're right.